l10n_eu_product_adr/models/product_template.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 Iryna Vyshnevska (Camptocamp)
+# Copyright 2021 Opener B.V. <stefan@opener.amsterdam>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    is_dangerous = fields.Boolean(help="This product belongs to a dangerous class")
+    adr_goods_id = fields.Many2one("adr.goods", "Dangerous Goods")
+    adr_class_id = fields.Many2one(
+        "adr.class", related="adr_goods_id.class_id", readonly=True
+    )
+    adr_classification_code = fields.Char(
+        related="adr_goods_id.classification_code", readonly=True
+    )
+    adr_label_ids = fields.Many2many(
+        "adr.label", related="adr_goods_id.label_ids", readonly=True
+    )
```

Comment:
```suggestion
    )
    limited_quantity = fields.Boolean("Limited Qty", compute="_compute_limited_quantity", store=True)
```
With the compute method like so
```python
@api.onchange("adr_goods_id", "adr_goods_id.limited_quantity_uom", "adr_goods_id.limited_quantity", "uom_id", )
def _compute_limited_quantity(self):
    for record in self:
        compare product's uom_qty with adr.goods uom_qty:
        product.limited_qty = adr.good uom_qty < product uom_qty
```

We should also ensure than the `adr.label` uom has the same uom family as the product one.

Reply:
I'm sorry, but my customer is a bulk trader. They trade in pallets, or gallons, and the individual subpackaging is encoded in the product name rather than the UoM of the product. So what suits your requirements is not generally valid.